### PR TITLE
ci: improve formatting checks

### PR DESCRIPTION
**Pull request by kevgo**
CI and CLI now always use the same version of Prettier and its shared config: the one in `package.json`. CI now runs `make format` instead of just Prettier, i.e. it now also runs `gofmt` and the `goimports` checker. To verify formatting, CI now formats the code and checks for uncommitted changes afterwards. This provides two benefits: 

1. CI can show a diff of the changes needed to make it pass. 
   This is the information needed to fix formatting problems manually.
   This allows us to not commit fixes on CI when submitting tiny documentation updates via the GitHub UI.
2. We can get rid of the `npm format:check` script, which was really only useful on CI.
   This script was highly redundant with `npm format`.
   Developers now simply run `make format` when they are unsure whether their code is formatted.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

**Comment by kevgo** [comment]
format

**Comment by kevgo** [comment]
format

**Comment by kevgo** [comment]
format

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,25 +25,32 @@ jobs:
...
+          path: node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json') }}
       - uses: actions/setup-go@v2
         with:
           go-version: ^1.17
+      - run: npm ci
+        if: steps.cache-node.outputs.cache-hit != 'true'
+      - run: make format
+      - name: Indicate formatting problems
+        run: git diff HEAD --exit-code
```
Comment (kevgo) [inline .github/workflows/ci.yml:50]: I was considering to extract this into a shared action in https://github.com/ory/ci but I think being explicit is better here.

**Comment by kevgo** [comment]
Closing and reopening to trigger the new format check to run.

**Comment by kevgo** [comment]
@aeneasr okay, I have fixed everything. Initially I followed the old setup but the `CI` status check creates uncommitted files on CI which leads to false positives in the format check. Long story short, the format check is now in a dedicated status check called `Format` and everything works.

While closing out this PR, can you please also remove the old required status checks that no longer exist (see attached screenshot)? Also, maybe make some of the existing status checks required again?

Screenshot from 2022-07-14 18-18-24

**Comment by aeneasr** [comment]
thanks!

**Pull request merged**